zxing-android-embedded/src/com/google/zxing/client/android/BeepManager.java

```diff
@@ -125,6 +122,7 @@ public boolean onError(MediaPlayer mp, int what, int extra) {
             return mediaPlayer;
         } catch (IOException ioe) {
             Log.w(TAG, ioe);
+            mediaPlayer.reset();
             mediaPlayer.release();
```

Comment:
Calling reset before release should fix the following warning: `W/MediaPlayer: mediaplayer went away with unhandled events`